db/migrations/20230227122704-ss58-prefix.js

```diff
@@ -0,0 +1,135 @@
+// Migration script for featured, era_commission, era_points, era_relative_performance, era_self_stake, era_vrc_score tables
+const convertEraTables = (db, ss58Format) => {
+    const tables = [
+        'featured',
+        'era_commission',
+        'era_points',
+        'era_relative_performance',
+        'era_self_stake',
+        'era_vrc_score',
+    ];
+    tables.forEach((table) => {
+        db.runSql(`SELECT * from ${table}`, (_, result) => {
+
+            result.rows.forEach(({stash_address}) => {
+
+                const nextAddress = keyring.encodeAddress(
+                    keyring.decodeAddress(stash_address),
+                    ss58Format);
+
+                const queryString = `UPDATE ${table} SET stash_address='${nextAddress}' WHERE stash_address='${stash_address}';`
+
+                db.runSql(queryString);
+
+            });
+
+        });
+    });
+};
+
+const convertRankingTable = (db, ss58Format) => {
+    db.runSql('SELECT * from ranking;', (_, result) => {
+
+        result.rows.forEach(({identity, stash_address, controller_address}) => {
+            const nextStashAddress = keyring.encodeAddress(
+                keyring.decodeAddress(stash_address),
+                ss58Format);
+
+            const nextControllerAddress = keyring.encodeAddress(
+                keyring.decodeAddress(controller_address),
+                ss58Format);
+
+            // TODO: Think how to deal with identity? Regex for addresses
+            const nextIdentity = identity;
+
+            const queryString = `UPDATE ranking SET identity='${nextIdentity}', stash_address='${nextStashAddress}', controller_address='${nextControllerAddress}' WHERE stash_address='${stash_address}';`
+
+            db.runSql(queryString);
+        });
+
+    });
+}
+
+// This batching approach is very slow, it should be replaced with creating temp. table and joining https://stackoverflow.com/questions/35903375/how-to-update-large-table-with-millions-of-rows-in-sql-server
+const convertBlockTable = (db, ss55Format) => {
+    db.runSql('SELECT block_author from block;', (_, result) => {
+        const totalRows = result.rows.length;
+        const batchSize = 10000;
+        let currentIndex = 0;
+
+        const updateBlock = (block_author) => {
+            const nextBlockAuthor = keyring.encodeAddress(
+                keyring.decodeAddress(block_author),
+                ss55Format);
+
+            const queryString = `UPDATE block SET block_author='${nextBlockAuthor}' WHERE block_author='${block_author}';`
+            db.runSql(queryString);
+        };
+
+        while (currentIndex < totalRows) {
+            const currentBatch = result.rows.slice(currentIndex, currentIndex + batchSize);
+
+            currentBatch.forEach(({block_author}) => {
+                updateBlock(block_author);
+            });
+
+            currentIndex += batchSize;
+        }
+    });
+};
+
+const convertTables = (db, prefix) => {
+    convertAccountTable(db, prefix);
```

Comment:
Can somebody write to/read from DC while migration is executing?

Reply:
I'm not sure if tables are accessible or if we can write data during the migration. Anyway we are going to stop crawlers and add maintainers mode during the migration.

Reply:
We need to check. The API is public so we do not control all clients.

Reply:
Ok, I'll check it

Reply:
I checked it. Basically, our migration is a SQL script wrapped into a transaction so we should check it from the transaction point of view.

1. **Reading**. There is no information about the transaction that I cannot make `select` query during the transaction. Also, I tried to run transactions up and down and read the data from the database during it and it works

2. **Writing** . There is a [quote](https://www.postgresql.org/docs/current/tutorial-transactions.html) from Postgres transaction docs: _The updates made so far by an open transaction are invisible to other transactions until the transaction completes, whereupon all the updates become visible simultaneously._ This means that during the transaction if we had to create a new account, it wouldn't affect the current migration and also of course new account will be added after the migration.